File: chroma-league/src/main/java/net/booone/chromaleague/hud/animations/ActivePlayerAssistAnimation.java
package net.booone.chromaleague.hud.animations;

import net.booone.chromaleague.hud.parts.AssistKillingSpreeBar;
import net.booone.chromaleague.hud.parts.Background;
import net.booone.razersdk.animation.AnimatedFrame;
import net.booone.razersdk.animation.LayeredFrame;
import net.booone.razersdk.animation.SimpleFrame;
import net.booone.razersdk.color.StaticColor;
import net.booone.razersdk.sdk.RzKey;

import java.util.List;

import static net.booone.razersdk.sdk.RzKey.*;

public class ActivePlayerAssistAnimation extends AnimatedFrame {

    public ActivePlayerAssistAnimation() {
        for (int i = 0; i < 4; i++) {
            addAnimationFrame(2, withBackground(List.of(RZKEY_O, RZKEY_L, RZKEY_DOT)));
            addAnimationFrame(2, withBackground(List.of(RZKEY_K, RZKEY_L, RZKEY_SEMICOLON)));
            addAnimationFrame(2, withBackground(List.of(RZKEY_M, RZKEY_L, RZKEY_SQUARE_BRACKET_LEFT)));
            addAnimationFrame(2, withBackground(List.of(RZKEY_COMA, RZKEY_L, RZKEY_P)));
        }
    }

    private static LayeredFrame withBackground(List<RzKey> rzKeys) {
        final LayeredFrame layeredFrame = new LayeredFrame();
        layeredFrame.addFrame(new SimpleFrame(AssistKillingSpreeBar.getKillingSpreeBar(), Background.getCurrentBackgroundColor()));
        layeredFrame.addFrame(new SimpleFrame(rzKeys, StaticColor.YELLOW));
        return layeredFrame;
    }
}

File: src/hud/animations/ActivePlayerAssistAnimation.ts
import {AnimatedFrame} from '../../razer-sdk/animation/AnimatedFrame.js';
import {LayeredFrame} from '../../razer-sdk/animation/LayeredFrame.js';
import {SimpleFrame} from '../../razer-sdk/animation/SimpleFrame.js';
import {StaticColor} from '../../razer-sdk/color/StaticColor.js';
import {RzKey} from '../../razer-sdk/sdk/RzKey.js';
import {getKillingSpreeBar} from '../parts/AssistKillingSpreeBar.js';
import {getCurrentBackgroundColor} from '../parts/Background.js';

export class ActivePlayerAssistAnimation extends AnimatedFrame {
    constructor() {
        super();
        for (let i = 0; i < 4; i++) {
            this.addAnimationFrame(2, withBackground([RzKey.RZKEY_O, RzKey.RZKEY_L, RzKey.RZKEY_DOT]));
            this.addAnimationFrame(2, withBackground([RzKey.RZKEY_K, RzKey.RZKEY_L, RzKey.RZKEY_SEMICOLON]));
            this.addAnimationFrame(2, withBackground([RzKey.RZKEY_M, RzKey.RZKEY_L, RzKey.RZKEY_SQUARE_BRACKET_LEFT]));
            this.addAnimationFrame(2, withBackground([RzKey.RZKEY_COMA, RzKey.RZKEY_L, RzKey.RZKEY_P]));
        }
    }
}

function withBackground(rzKeys: RzKey[]): LayeredFrame {
    const layeredFrame = new LayeredFrame();
    layeredFrame.addFrame(new SimpleFrame(getKillingSpreeBar(), getCurrentBackgroundColor()));
    layeredFrame.addFrame(new SimpleFrame(rzKeys, StaticColor.YELLOW));
    return layeredFrame;
}
